src/Controller/CheckoutRequestController.php

```diff
@@ -252,4 +254,24 @@ protected function processException($errorType, $errorMessage, \Exception $excep
 
         return $this->getResponse()->withJson($errorResp)->withStatus($statusCode);
     }
+
+    /**
+     * @param array $data should have a patronBarcode attribute, which will be randomly reassigned for partners
+     * @return array will be a copy of the input with patronBarcode reassigned in case it was originally a partner barcode
+     */
+    public function reassignPartnerBarcode($data)
+    {
+      $key = "PATRON_BARCODES_{$data['patronBarcode']}";
+      $barcodes = explode("," , Config::get($key, "", true));
+      $barcodes = array_filter($barcodes, function ($item) { return $item; });
+      $numberOfPatronBarcodes = count($barcodes);
+      if ($numberOfPatronBarcodes >= 1) {
+        $newBarcode = $barcodes[rand(0, $numberOfPatronBarcodes - 1)];
+        if ($newBarcode) {
```

Comment:
I like the check here to see if the data really exits! But I think a better place will be line #265. Because if for any reason `Config::get` failed, it will return `null` by default, or since we put `""` here, it will now return `""`.  And then it will throw warning: ` Undefined offset: 1 in...` from `explode ()`

Reply:
If `Config::get` falls back on the default, we'll run `explode` on an empty string (or `null` if we change that to `null`). In either case, `$barcodes` will be an array of 0 - N length. Only if length is greater than 0 will we attempt to reach into the array by offset. Having trouble seeing where the error might be thrown.. Where might an `Undefined offset` error be thrown from?